internal/service/databaseuser/resource_database_user_test.go

```diff
@@ -494,6 +495,52 @@ func TestAccConfigRSDatabaseUser_withLDAPAuthType(t *testing.T) {
 	})
 }
 
+func TestAccCOnfigRSDatabaseUser_withOIDCAuthType(t *testing.T) {
+	var (
+		projectID         = acc.ProjectIDExecution(t)
+		idpID             = os.Getenv("MONGODB_ATLAS_FEDERATED_IDP_ID")
+		workforceAuthType = "IDP_GROUP"
+		workloadAuthType  = "USER"
+		usernameWorkforce = fmt.Sprintf("%s/%s", idpID, workforceAuthType)
+		usernameWorkload  = fmt.Sprintf("%s/%s", idpID, workloadAuthType)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: acc.ConfigDataBaseUserWithOIDCAuthType(projectID, usernameWorkforce, workforceAuthType, "admin", "atlasAdmin"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
+					resource.TestCheckResourceAttr(resourceName, "username", usernameWorkforce),
+					resource.TestCheckResourceAttr(resourceName, "oidc_auth_type", workforceAuthType),
+					resource.TestCheckResourceAttr(resourceName, "auth_database_name", "admin"),
+				),
+			},
+			{
+				Config: acc.ConfigDataBaseUserWithOIDCAuthType(projectID, usernameWorkload, workloadAuthType, "$external", "atlasAdmin"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
+					resource.TestCheckResourceAttr(resourceName, "username", usernameWorkload),
+					resource.TestCheckResourceAttr(resourceName, "username", usernameWorkload),
+					resource.TestCheckResourceAttr(resourceName, "auth_database_name", "$external"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       importStateIDFunc(resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"password"},
```

Comment:
curious why password is ignored?

Reply:
In `TestAccConfigRSDatabaseUser_basic` test it is necessary, not in this one. I removed it where it was unnecessary. 
The reason of `ImportStateVerifyIgnore` is because password is not returned in the API call, so if we import the resource and in the configuration we have set the value of password, there will be a mismatch. That is why it has to be ignored.